### PR TITLE
k8s 1.27 support for helm installation

### PR DESCRIPTION
**Pull request by bandak2**
# Description
Add changes to support k8s 1.27

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/761 |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
installation test, sanity test
image

image

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Review by rensyct — APPROVED** [review]

File: dell-csi-helm-installer/verify-csi-unity.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 # verify-csi-unity method
 function verify-csi-unity() {
-  verify_k8s_versions "1.22" "1.26"
+  verify_k8s_versions "1.22" "1.27"
```
Comment (karthikk92) [inline dell-csi-helm-installer/verify-csi-unity.sh:13]: So do we need to support k8s 1.22 ?

Comment (bandak2) [reply dell-csi-helm-installer/verify-csi-unity.sh:13]: yes, until Tanzu support can be confirmed. Once the licensing gets processed and we can confirm having a 1.22+ config there, I believe we can remove support for 1.22 here.

File: helm/csi-unity/Chart.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 name: csi-unity
 version: 2.7.0
 appVersion: 2.7.0
-kubeVersion: ">= 1.24.0 < 1.27.0"
+kubeVersion: ">= 1.25.0 < 1.28.0"
```
Comment (panigs7) [inline helm/csi-unity/Chart.yaml:4]: Hi @bandak2 should the minimum version be 1.22?

Comment (bandak2) [reply helm/csi-unity/Chart.yaml:4]: yes, until our Tanzu support for 1.22+ can be confirmed.

Comment (bandak2) [reply helm/csi-unity/Chart.yaml:4]: This check would work with standard k8s versions, and validation would happen to support 1.25 through 1.27. For other k8s versions whose validation string is different, the user would have to comment the above line in Chart.yaml and uncomment line

`# kubeVersion: ">= 1.22.0-0 < 1.28.0-0"`

This is in line and noted in csi-docs
https://dell.github.io/csm-docs/docs/csidriver/installation/helm/unity/

> If the user is using a complex K8s version like “v1.24.6-mirantis-1”, use this kubeVersion check in helm/csi-unity/Chart.yaml file. 
kubeVersion: ">= 1.24.0-0 < 1.27.0-0"

**Review by panigs7 — APPROVED** [review]

**Review by karthikk92 — APPROVED** [review]

**Pull request merged**